### PR TITLE
Update deprecation/removal docs to point to labels/milestones instead

### PR DESCRIPTION
**Pull request by nicoddemus**
Using milestones and proper issues are a much saner way to handle these topics
than keeping them in sync in a separate document.

I was worried how to keep them in sync until this solution occurred to me. I already created the labels and related issues.

### CODE REVIEW

File: doc/en/backwards-compatibility.rst [hunk 1]
```diff
@@ -15,91 +15,6 @@ We will only remove deprecated functionality in major releases (e.g. if we depre
 Deprecation Roadmap
 -------------------
 
-This page lists deprecated features and when we plan to remove them. It is important to list the feature, the version where it got deprecated and the version we plan to remove it.
+We track deprecation and removal of features using milestones and the `deprecation <https://github.com/pytest-dev/pytest/labels/type%3A%20deprecation>`_ and `removal <https://github.com/pytest-dev/pytest/labels/type%3A%20removal>`_ labels on GitHub.
```
Comment (asottile) [inline doc/en/backwards-compatibility.rst]: should we perhaps link to the label page which shows both open and closed? (for example https://github.com/pytest-dev/pytest/issues?q=label%3A%22type%3A+deprecation%22)

Comment (nicoddemus) [reply doc/en/backwards-compatibility.rst]: Good idea, done!

**Comment by hugovk** [comment]
The [label for `deprecation`](https://github.com/pytest-dev/pytest/labels?utf8=%E2%9C%93&q=deprecation+removal) says:

> feature that will be deprecated in the future

I think the feature is being deprecated _now_, would it make more sense to say this?

> feature that will be removed in the future

**Comment by nicoddemus** [comment]
@hugovk good point, fixed. Thanks!

**Comment by nicoddemus** [comment]
Awesome, thanks for the feedback, we appreciate it!

**Pull request merged**